src/url-params.ts

```diff
@@ -63,12 +63,12 @@ export class AnalyticsUrlParams {
 				passedDeviceId,
 				originalDeviceId,
 			);
-			if (this.client != null && newCurrentDeviceId != null) {
-				this.client.setDeviceId(newCurrentDeviceId);
-			}
+			//if (this.client != null && newCurrentDeviceId != null) {
+			//	this.client.setDeviceId(newCurrentDeviceId);
+			//}
 
 			params.delete(URL_PARAM_DEVICE_ID);
-			return params.toString();
+			return [passedDeviceId, params.toString()]; //Im passing `passedDeviceId` cause I dont know how to set the typescript return type in order to accept a list as returned by setDeviceIds()
```

Comment:
`setDeviceIds` function stores the passed device identifiers under a cookie and we store all the values under the private property `deviceIds`. So why does this function need to return the parsed values back?

Reply:
Right, my mistake. I thought setDeviceIds returned the whole array used to write the cookie values but indeed it returns only the first element of the array, which should be the passed value.

But the reason I changed it to return the parsed value was to then user it for Client init, without having to add the logic of reading the passed value from directly from the cookie later on.